lib/udev/rules.d/90-revpi-bluetooth.rules

```diff
@@ -0,0 +1,5 @@
+# Revolution Pi bluetooth module: start the hciuart service
+ACTION=="add", DEVPATH=="*/usb1/1-1/1-1.5/1-1.5:1.0/*", TAG+="systemd", ENV{SYSTEMD_WANTS}+="revpi-hciuart@%k.service"
```

Comment:
does disabling or masking the service still work?

Reply:
by disable it is still work after rebooting, which means it is always automatically started by the udev rules

by stop it is "dead", and "hcitool dev " show nothing, seems as the "hciattach ...." never been executed.

Reply:
Please start giving some more context. The way you are giving information doesn't help. Please think about the other person is not involved in this like you are. And be precise when giving information.

I don't know what you want to tell with this.

Reply:
pi@RevPi123456:~ $ systemctl status revpi-hciuart@ttyUSB0
● revpi-hciuart@ttyUSB0.service - Revolution Pi Bluetooth Device ttyUSB0
   Loaded: loaded (/lib/systemd/system/revpi-hciuart@.service; static; vendor preset: enabled)
   Active: active (running) since Thu 2021-03-11 10:41:09 CET; 4min 11s ago
  Process: 3292 ExecStart=/usr/bin/revpi-btuart ttyUSB0 (code=exited, status=0/SUCCESS)
 Main PID: 3320 (hciattach)
    Tasks: 1 (limit: 2199)
   Memory: 472.0K
   CGroup: /system.slice/system-revpi\x2dhciuart.slice/revpi-hciuart@ttyUSB0.service
           └─3320 /usr/bin/hciattach ttyUSB0 any

Mar 11 10:41:09 RevPi123456 systemd[1]: Starting Revolution Pi Bluetooth Device ttyUSB0...
Mar 11 10:41:09 RevPi123456 revpi-btuart[3292]: Device setup complete
Mar 11 10:41:09 RevPi123456 systemd[1]: Started Revolution Pi Bluetooth Device ttyUSB0.

pi@RevPi123456:~ $ hcitool dev
Devices:
        hci0    B8:27:EB:51:41:69
pi@RevPi123456:~ $
pi@RevPi123456:~ $ sudo systemctl disable revpi-hciuart@ttyUSB0
pi@RevPi123456:~ $ systemctl status revpi-hciuart@ttyUSB0
● revpi-hciuart@ttyUSB0.service - Revolution Pi Bluetooth Device ttyUSB0
   Loaded: loaded (/lib/systemd/system/revpi-hciuart@.service; static; vendor preset: enabled)
   Active: active (running) since Thu 2021-03-11 10:41:09 CET; 4min 55s ago
 Main PID: 3320 (hciattach)
    Tasks: 1 (limit: 2199)
   Memory: 472.0K
   CGroup: /system.slice/system-revpi\x2dhciuart.slice/revpi-hciuart@ttyUSB0.service
           └─3320 /usr/bin/hciattach ttyUSB0 any

Mar 11 10:41:09 RevPi123456 systemd[1]: Starting Revolution Pi Bluetooth Device ttyUSB0...
Mar 11 10:41:09 RevPi123456 revpi-btuart[3292]: Device setup complete
Mar 11 10:41:09 RevPi123456 systemd[1]: Started Revolution Pi Bluetooth Device ttyUSB0.

pi@RevPi123456:~ $ sudo reboot

pi@RevPi123456:~ $ systemctl status revpi-hciuart@ttyUSB0
● revpi-hciuart@ttyUSB0.service - Revolution Pi Bluetooth Device ttyUSB0
   Loaded: loaded (/lib/systemd/system/revpi-hciuart@.service; static; vendor preset: enabled)
   Active: active (running) since Thu 2021-03-11 10:46:55 CET; 13s ago
  Process: 3292 ExecStart=/usr/bin/revpi-btuart ttyUSB0 (code=exited, status=0/SUCCESS)
 Main PID: 3327 (hciattach)
    Tasks: 1 (limit: 2199)
   Memory: 428.0K
   CGroup: /system.slice/system-revpi\x2dhciuart.slice/revpi-hciuart@ttyUSB0.service
           └─3327 /usr/bin/hciattach ttyUSB0 any

Mar 11 10:46:55 RevPi123456 systemd[1]: Starting Revolution Pi Bluetooth Device ttyUSB0...
Mar 11 10:46:55 RevPi123456 revpi-btuart[3292]: Device setup complete
Mar 11 10:46:55 RevPi123456 systemd[1]: Started Revolution Pi Bluetooth Device ttyUSB0.

Reply:
pi@RevPi123456:~ $ systemctl status revpi-hciuart@ttyUSB0
● revpi-hciuart@ttyUSB0.service - Revolution Pi Bluetooth Device ttyUSB0
   Loaded: loaded (/lib/systemd/system/revpi-hciuart@.service; static; vendor preset: enabled)
   Active: active (running) since Thu 2021-03-11 10:46:55 CET; 13s ago
  Process: 3292 ExecStart=/usr/bin/revpi-btuart ttyUSB0 (code=exited, status=0/SUCCESS)
 Main PID: 3327 (hciattach)
    Tasks: 1 (limit: 2199)
   Memory: 428.0K
   CGroup: /system.slice/system-revpi\x2dhciuart.slice/revpi-hciuart@ttyUSB0.service
           └─3327 /usr/bin/hciattach ttyUSB0 any

Mar 11 10:46:55 RevPi123456 systemd[1]: Starting Revolution Pi Bluetooth Device ttyUSB0...
Mar 11 10:46:55 RevPi123456 revpi-btuart[3292]: Device setup complete
Mar 11 10:46:55 RevPi123456 systemd[1]: Started Revolution Pi Bluetooth Device ttyUSB0.
pi@RevPi123456:~ $ hcitool dev
Devices:
hci0 B8:27:EB:51:41:69
pi@RevPi123456:~ $
pi@RevPi123456:~ $
pi@RevPi123456:~ $
pi@RevPi123456:~ $
pi@RevPi123456:~ $ sudo systemctl stop revpi-hciuart@ttyUSB0
pi@RevPi123456:~ $ systemctl status revpi-hciuart@ttyUSB0
● revpi-hciuart@ttyUSB0.service - Revolution Pi Bluetooth Device ttyUSB0
   Loaded: loaded (/lib/systemd/system/revpi-hciuart@.service; static; vendor preset: enabled)
   Active: inactive (dead) since Thu 2021-03-11 10:50:31 CET; 4s ago
  Process: 3292 ExecStart=/usr/bin/revpi-btuart ttyUSB0 (code=exited, status=0/SUCCESS)
 Main PID: 3327 (code=exited, status=0/SUCCESS)

Mar 11 10:46:55 RevPi123456 systemd[1]: Starting Revolution Pi Bluetooth Device ttyUSB0...
Mar 11 10:46:55 RevPi123456 revpi-btuart[3292]: Device setup complete
Mar 11 10:46:55 RevPi123456 systemd[1]: Started Revolution Pi Bluetooth Device ttyUSB0.
Mar 11 10:50:29 RevPi123456 systemd[1]: Stopping Revolution Pi Bluetooth Device ttyUSB0...
Mar 11 10:50:31 RevPi123456 systemd[1]: revpi-hciuart@ttyUSB0.service: Succeeded.
Mar 11 10:50:31 RevPi123456 systemd[1]: Stopped Revolution Pi Bluetooth Device ttyUSB0.
pi@RevPi123456:~ $ hcitool dev
Devices:

Reply:
Ok most of this is like expacted. AFAIK when a service is disabled the service is not stopped. And a disabled service can be started.

That's why I asked about masking the service.

Reply:
got that, the masking makes it not start:
pi@RevPi123456:~ $ hcitool dev
Devices:
        hci0    B8:27:EB:51:41:69
pi@RevPi123456:~ $ sudo ^Cstemctl mask
pi@RevPi123456:~ $ sudo systemctl mask revpi-hciuart@ttyUSB0
Created symlink /etc/systemd/system/revpi-hciuart@ttyUSB0.service → /dev/null.
pi@RevPi123456:~ $ sudo reboot

pi@RevPi123456:~ $ systemctl status revpi-hciuart@ttyUSB0
● revpi-hciuart@ttyUSB0.service
   Loaded: masked (Reason: Unit revpi-hciuart@ttyUSB0.service is masked.)
   Active: inactive (dead)
pi@RevPi123456:~ $ hcitool dev
Devices:
pi@RevPi123456:~ $

Reply:
OK this should be ok